tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -2758,6 +2758,10 @@ port::Status MIOpenSupport::DoPrepareForConvolution(
     wrap::miopenSetAllocator(miopen.handle(), nullptr, nullptr, nullptr);
 
     scratch_memory_size = preference.memory;
+
+    // Reuse the scratch_memory_temp in actual convolution
```

Comment:
add an assert here to verify the assumption we are making here, i.e. `scratch_memory_size <= size_in_bytes`.  always holds

Reply:
Sounds good